var/lib/delphix-platform/ansible/10-delphix-platform/roles/delphix-platform/tasks/main.yml

```diff
@@ -294,3 +294,41 @@
     #
     - { key: 'AutoUpdate.Enabled', value: 'n' }
   when: platform == "azure"
+
+#
+# Customize the GCP linux environment.
+#
+# Update the override file for the GCP instance. This file gets
+# applied dynamically by running google_instance_setup script.
+#
+- blockinfile:
+    path: /etc/default/instance_configs.cfg.template
+    create: yes
+    block: |
+      #
+      # Disable the accounts daemon to prevent adding/removing
+      # users on the engine.
+      #
+      [Daemons]
+      accounts_daemon = false
+
+      #
+      # Disable user supplied startup/shutdown scripts from running on
+      # the engine.
+      #
+      [MetadataScripts]
+      shutdown = false
+      startup = false
+  when:
+    - platform == "gcp"
+    - not ansible_is_chroot
+  notify: "gcp config changed"
+
+#
+# We don't want to run the account daemon when we first boot up so
+# disable here.
+#
+- command: systemctl disable google-accounts-daemon.service
+  when:
+    - platform == "gcp"
+    - ansible_is_chroot
```

Comment:
you're doing this, so we only run this command during the build, and not during the first boot? even though this is possible, I'm not sure I think we should be relying on "ansible_is_chroot" like this..

what are the requirements here? I wonder if we should disable this via the post-install package hook instead? why don't we want to run this disable anytime the platform service runs?

I just worry that if we start to do this, we'll end up with lots of conditional logic (e.g. "do this during the build, but not during runtime", "do this during runtime, but not during the build", etc.), making things more complicated for little benefit (or when a better approach exists). The original idea is we wouldn't have any build/runtime conditionas; the only reason I think using "ansible_is_chroot" makes sense, is when logic we'd run won't work during the build (e.g. creating zfs datasets).

Reply:
We could do this all the time so I can remove the `ansible_is_chroot`. Is the post-install hook a preferred way? I'm not that familiar with that but am willing to change it.

Reply:
if we're OK always disabling the service any time the delphix-platform service runs, then keeping it here and removing the "ansible_is_chroot" flag is best, I think.

Ultimately, I think it comes down to deciding if we want to:

1. Disable the service once and only once. This way, if the service is later enabled, and the platform service re-run, the service stays enabled.

2. Or, ensure the service is always disabled. So, if the service is later enabled, the platform service will "fix" that by disabling the service the next time it runs.

If we remove the "ansible_is_chroot" part, I feel like we want (2), so I think what you have here is best.

Reply:
I view the ansible code as a declaration of how we want the system to configured, rather than a way of executing commands.. and if we could use the "service" module, this difference would be more clear.

So, in this case, we're stating we always want this service to be disabled, rather than a way of running "systemctl disable".